src/utils/polygonUtils.cpp

```diff
@@ -1251,7 +1251,7 @@ bool PolygonUtils::polygonCollidesWithLineSegment(ConstPolygonRef poly, const Po
 
 bool PolygonUtils::polygonCollidesWithLineSegment(const Polygons& polys, const Point& transformed_startPoint, const Point& transformed_endPoint, PointMatrix transformation_matrix)
 {
-    for (ConstPolygonRef poly : polys)
+    for (const ConstPolygonRef poly : polys)
```

Comment:
Not really a comment on this change specific;  Just a remark on the strangeness which is CuraEngine: "Why do we even have those strange specialization for the polygon-class?" 
@rburema we should have a propper look at this class, together with @nallath

Reply:
Eventually. Let's drill down until we got there. First we have to have that session about the architecture ;-)

Maybe there should be a place to collate these whichlist/fixme/todo's.